### PR TITLE
SW-2746 Order weight units according to preference

### PR DESCRIPTION
**Pull request by constanzauanini**
Imperial:
Screenshot 2023-01-25 at 15 21 19
Screenshot 2023-01-25 at 15 21 50

Metric:
Screenshot 2023-01-25 at 15 22 33
Screenshot 2023-01-25 at 15 22 12

### CODE REVIEW

File: src/components/accession2/withdraw/WithdrawModal.tsx [hunk 1]
```diff
@@ -464,7 +464,7 @@ export default function WithdrawDialog(props: WithdrawDialogProps): JSX.Element
                 <Box>{strings.CT}</Box>
               ) : (
                 <Dropdown
-                  options={weightUnitsV2()}
+                  options={weightUnitsV2(userPreferences.preferredWeightSystem as string)}
```
Comment (karthikbtf) [inline src/components/accession2/withdraw/WithdrawModal.tsx:467]: suggestion, create another function in units.ts which returns weightUnitsV2 based off user preference.. we can avoid some repetition code here

**Review by karthikbtf — APPROVED** [review]
one suggestion

**Review by karthikbtf — APPROVED** [review]
Sorry for not being clear in my last review.. added some clarification.
Feel free to merge as-is if you prefer how it looks.

File: src/units.ts [hunk 2]
```diff
@@ -34,6 +35,10 @@ export function weightUnitsV2(unit?: string) {
   ];
 }
 
+export function weightUnitsOrderedByPreference(userPreferences: PreferencesType) {
```
[the remark this answers is not in the document]
Comment (constanzauanini) [reply src/units.ts:38]: Oh, thanks for this suggestion!

**Review by karthikbtf — APPROVED** [review]
Thanks 🎉

File: src/units.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (karthikbtf) [reply src/units.ts]: Looks good,,, is the suggestion working correctly? in case i made a mistake..

[the remark this answers is not in the document]
Comment (constanzauanini) [reply src/units.ts]: It is working!

**Pull request merged**